src/silicone/database_crunchers/equal_quantile_walk.py

```diff
@@ -0,0 +1,157 @@
+
+            Returns
+            -------
+            :obj:`pyam.IamDataFrame`
+                Filled in data (without original source data)
+
+            Raises
+            ------
+            ValueError
+                The key year for filling is not in ``in_iamdf`` and ``interpolate is
+                False``.
+            """
+            lead_in = in_iamdf.filter(variable=variable_leaders)
+            if not all(lead_in.variables(True)["unit"] == lead_unit):
+                raise ValueError(
+                    "Units of lead variable is meant to be `{}`, found `{}`".format(
+                        lead_unit, lead_in.variables(True)["unit"].tolist()
+                    )
+                )
+
+            if data_follower_time_col != in_iamdf.time_col:
+                raise ValueError(
+                    "`in_iamdf` time column must be the same as the time column used "
+                    "to generate this filler function (`{}`)".format(
+                        data_follower_time_col
+                    )
+                )
+            if lead_in.data.empty:
+                raise ValueError(
+                    "There is no data for {} so it cannot be infilled".format(
+                        variable_leaders
+                    )
+                )
+            output_ts = lead_in.timeseries()
+            if any(
+                [
+                    (time not in lead_ts.columns) or (time not in follower_ts.columns)
+                    for time in output_ts.columns
+                ]
+            ):
+                # We allow for cases where either lead or follow have gaps
+                raise ValueError(
+                    "Not all required timepoints are present in the database we "
+                    "crunched, we crunched \n\t{} for the lead and \n\t{} for the "
+                    "follow \nbut you passed in \n\t{}".format(
+                        lead_ts.columns, follower_ts.columns, output_ts.columns
+                    )
+                )
+            for col in output_ts.columns:
+                output_ts[col] = self._find_same_quantile(
+                    follower_ts[col], lead_ts[col], output_ts[col]
+                )
+            output_ts = output_ts.reset_index()
+            output_ts["variable"] = variable_follower
+            output_ts["unit"] = data_follower_unit
+            return IamDataFrame(output_ts)
+
+        return filler
+
+    def _get_iamdf_follower(self, variable_follower, variable_leaders):
+        if len(variable_leaders) > 1:
+            raise ValueError(
+                "For `EqualQuantileWalk`, ``variable_leaders`` should only "
+                "contain one variable"
+            )
+
+        self._check_follower_and_leader_in_db(variable_follower, variable_leaders)
+
+        return self._db.filter(variable=variable_follower)
+
+    def _find_same_quantile(self, follow_vals, lead_vals, lead_input):
+        if len(lead_vals) == 1:
```

Comment:
```suggestion
        if len(follow_vals) == 1:
```

Would this be clearer?

Reply:
It's to avoid a singularity later, we could also short-circuit the calculation in the case of 1 follow but that's for computational reasons not essential ones. I guess it's better to take the mean afterwards in case the length of the two are different.